### ISSUE TITLE
Alert e-mails no longer contain detailed NACK reason/error information

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3390]
In 2.x when you receive a NACK, in the alert e-mail you receive you can see exactly why the message was rejected:

{quote}ERROR-408: MLLP Connector error
ERROR MESSAGE:	NACK sent from receiver:  \[Application Reject\]
Message Rejected for this specific reason.
: MSH|^~\&|SHMADT|SHM|ADT1|SHM|20141021081611||ACK^A01^ACK|20141021081611|P|2.5
MSA|AR|MSG00001|Message Rejected for this specific reason.{quote}
\\
However in 3.x, you don't get any of that helpful information:

{quote}Response Validation (TCP Sender) error
ERROR MESSAGE: NACK sent from receiver.{quote}

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3493
Reporter: narupley
Created: 2014-10-21T08:21:44.000-0700

**Comment by rbeckman-nextgen** [comment]
In 3.0.x there wouldn't even be an alert at all. That was fixed in 3.1.0 (MIRTH-3295) but it appears there is some information missing that would still be useful.

Imported Comment. Original Details:
Author: wayneh
Created: 2014-10-21T10:26:42.000-0700

**Comment by rbeckman-nextgen** [comment]
Adding patch

Imported Comment. Original Details:
Author: narupley
Created: 2014-11-03T11:35:57.000-0800

**Comment by rbeckman-nextgen** [comment]
Revisions 7443/7444: Added additional information to the response status when an HL7 NACK occurs, which will be available in alert actions. An indicator of rejected/error is included, as well as MSA.3 and ERR.1, if present.

Imported Comment. Original Details:
Author: narupley
Created: 2014-11-06T16:15:01.000-0800

**Comment by rbeckman-nextgen** [comment]
Verified that the reason is now included in the alert error message. One more potential issue is that there is currently no good way to identify which message the alert came from. One option is to add the entire ACK like what used to happen in 2.x, but we don't want to double store the response.

Therefore we have created MIRTH-3520 so we can add the message ID and potentially more information to the alerts.

Imported Comment. Original Details:
Author: wayneh
Created: 2014-11-11T12:07:05.000-0800
